examples/jsm/nodes/core/PropertyNode.js

```diff
@@ -27,6 +28,8 @@ class PropertyNode extends Node {
 
 	generate( builder ) {
 
+		if ( this.declare === false ) return this.name;
```

Comment:
Maybe we can create a separate node for this like `ReferenceNode` (current `ReferenceNode` we can merge with `MaterialReferenceNode` maybe)?

Reply:
`ReferenceNode` has other uses as used in `userData()`. The function signature looks good for me, mainly because the declaration comes after the name, which would be a fundamental parameter.